src/components/dialogs/network-modifications/tabular-modification/tabular-modification-dialog.js

```diff
@@ -0,0 +1,188 @@
+import ModificationDialog from 'components/dialogs/commons/modificationDialog';
+import { createTabulareModification } from 'services/study/network-modifications';
+import { FetchStatus } from 'services/utils';
+import TabularModificationForm from './tabular-modification-form';
+import {
+    TABULAR_MODIFICATION_FIELDS,
+    TABULAR_MODIFICATION_TYPES,
+    formatModification,
+    getEquipmentTypeFromModificationType,
+} from './tabular-modification-utils';
+import { toModificationOperation } from 'components/utils/utils';
+import { useIntl } from 'react-intl';
+
+const formSchema = yup
+    .object()
+    .shape({
+        [TYPE]: yup.string().nullable().required(),
+    })
+    .required();
+
+const emptyFormData = {
+    [TYPE]: null,
+    [MODIFICATIONS_TABLE]: [],
+};
+
+/**
+ * Dialog to create tabular modification based on a csv file.
+ * @param studyUuid the study we are currently working on
+ * @param currentNode the node we are currently working on
+ * @param editData the data to edit
+ * @param isUpdate check if edition form
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ * @param editDataFetchStatus indicates the status of fetching EditData
+ */
+const TabularModificationDialog = ({
+    studyUuid,
+    currentNode,
+    editData,
+    isUpdate,
+    editDataFetchStatus,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode?.id;
+
+    const intl = useIntl();
+
+    const { snackError } = useSnackMessage();
+
+    const formMethods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(formSchema),
+    });
+
+    const { reset, getValues } = formMethods;
+
+    const fromEditDataToFormValues = useCallback(
+        (editData) => {
+            const equipmentType = getEquipmentTypeFromModificationType(
+                editData?.modificationType
+            );
+            const modifications = editData?.modifications.map((modif) => {
+                let modification = {};
+                Object.getOwnPropertyNames(formatModification(modif)).forEach(
+                    (key) => {
+                        const translatedKey = intl.formatMessage({
+                            id: key,
+                        });
+                        if (key === 'equipmentId') {
+                            modification[translatedKey] = modif[key];
+                        } else if (typeof modif[key] === 'object') {
+                            modification[translatedKey] = modif[key]?.value;
+                        }
+                    }
+                );
+                return modification;
+            });
+            reset({
+                [TYPE]: equipmentType,
+                [MODIFICATIONS_TABLE]: modifications,
+            });
+        },
+        [intl, reset]
+    );
+
+    useEffect(() => {
+        if (editData) {
+            fromEditDataToFormValues(editData);
```

Comment:
you can move all the code inside the callback inside this useEffect and have [reset, editData, intl] as dependencies, it is the same thing.
this way you can avoid "fromEditDataToFormValues" creation.